src/components/settings/Password.js

```diff
@@ -49,53 +47,63 @@ class Password extends Component<Props, State> {
 
     if (oldPassword && newPassword && newPasswordConfirm) {
       if (newPassword !== newPasswordConfirm) {
-        this.handleMismatchedPassword();
+        this.handleError('mismatchedPassword', true);
+        return false;
+      }
+      if (oldPassword === newPassword) {
+        this.handleError('samePassword', true);
         return false;
       }
       dispatch(changePassword(user.id, oldPassword, newPassword));
     } else {
-      this.handleMissingInput();
+      this.handleError('missingInput', true);
     }
     return true;
   }
 
-  handleMissingInput = () => {
+  handleError = (name, value) => {
     this.setState({
-      missingInput: true,
+      errors: {
+        [name]: value,
+      },
     });
   }
 
-  handleMismatchedPassword = () => {
+  resetErrors = () => {
     this.setState({
-      mismatchedPassword: true,
+      errors: {},
     });
   }
 
-  resetErrors = () => {
-    this.setState({
-      mismatchedPassword: false,
-      missingInput: false,
-    });
+  printError = () => {
+    const errorMessages = {
+      mismatchedPassword: 'Passwords do not match.',
+      missingInput: 'Please fill in all fields.',
+      samePassword: 'Your new password cannot be the same as your old one.',
+    };
```

Comment:
Is there a clean way to extract this into a general `ValidationErrors` Component? Configurable with possible error and messages that you just pass the errors object as state? That way we'd remove some duplication between this and the register page. And we can also reuse it for other form validation errors if they ever come up.

Reply:
Sounds like a good idea!